heat/core/io.py

```diff
@@ -895,8 +895,14 @@ def load_csv(
         # In case there are some empty lines in the csv file
         local_tensor = local_tensor[:actual_length]
 
-        resulting_tensor = factories.array(
-            local_tensor, dtype=dtype, is_split=0, device=device, comm=comm
+        resulting_tensor = DNDarray(
+            local_tensor,
+            gshape=tuple(local_tensor.shape),
+            dtype=dtype,
+            split=0,
+            device=device,
+            comm=comm,
+            balanced=local_tensor.is_balanced,
```

Comment:
`local_tensor` is a `torch.Tensor`. It doesn't "know" about being a slice of a larger distributed array.

`is_balanced()` is a method of the `DNDarray` class, it has to do with whether the memory-distributed [DNDarray](https://heat.readthedocs.io/en/latest/autoapi/heat/core/dndarray/index.html) is distributed evenly among the available processes.

If we cannot asses load balance of the output DNDarray (I'm not sure that's the case here, I haven't checked), we set `balanced =  None`.